ILSpy/LoadedPackage.cs

```diff
@@ -197,7 +197,24 @@ public BundleEntry(string bundleFile, MemoryMappedViewAccessor view, SingleFileB
 			public override Stream TryOpenStream()
 			{
 				Debug.WriteLine("Open bundle member " + Name);
-				return new UnmanagedMemoryStream(view.SafeMemoryMappedViewHandle, entry.Offset, entry.Size);
+
+				if (entry.CompressedSize == 0)
+				{
+					return new UnmanagedMemoryStream(view.SafeMemoryMappedViewHandle, entry.Offset, entry.Size);
+				}
+				else
+				{
+					Stream compressedStream = new UnmanagedMemoryStream(view.SafeMemoryMappedViewHandle, entry.Offset, entry.CompressedSize);
+					using var deflateStream = new DeflateStream(compressedStream, CompressionMode.Decompress);
+					Stream decompressedStream = new MemoryStream((int)entry.Size);
+					deflateStream.CopyTo(decompressedStream);
+					if (decompressedStream.Length != entry.Size)
+					{
+						throw new InvalidDataException($"Corrupted single-file entry '${entry.RelativePath}'. Declared decompressed size '${entry.Size}' is not the same as actual decompressed size '${decompressedStream.Length}'.");
+					}
+
```

Comment:
Don't we have `decompressedStream.Position == decompressedStream.Length` at this point?
I think we need to seek back to the start before returning the stream.